plugin/evm/warp_client.go

```diff
@@ -0,0 +1,38 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/utils/crypto/bls"
+	"github.com/ava-labs/avalanchego/utils/rpc"
+	"github.com/ava-labs/subnet-evm/plugin/evm/message"
+)
+
+// Interface compliance
+var _ WarpClient = (*warpClient)(nil)
+
+type WarpClient interface {
+	GetSignature(ctx context.Context, signatureRequest message.SignatureRequest) (*[bls.SignatureLen]byte, error)
+}
+
+// Client implementation for interacting with EVM [chain]
+type warpClient struct {
+	requester rpc.EndpointRequester
+}
+
+// NewClient returns a Client for interacting with EVM [chain]
+func NewWarpClient(uri, chain string) WarpClient {
+	return &warpClient{
+		requester: rpc.NewEndpointRequester(fmt.Sprintf("%s/ext/bc/%s/warp", uri, chain)),
```

Comment:
nit: where is this endpoint from? Do we need the same chainID twice in the endpoint or does one work as well?

Reply:
The implementation has been updated since you left this comment, but the endpoint is the same as the eth rpc API, i.e. `/rpc`. Methods are denoted by `warp_methodName` in the `method` field.